File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.ts
function maxProduct(n: number): number {
    let first = 0, second = 0;
    while (n > 0) {
        const digit = n % 10;

        if (digit >= first) {
            second = first;
            first = digit;
        } else if (digit > second) {
            second = digit;
        }
        
        n = Math.floor(n / 10);
    }
    return first * second;
}

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.rs
impl Solution {
    pub fn max_product(n: i32) -> i32 {
        let mut n = n;
        let mut first = 0;
        let mut second = 0;
        while n > 0 {
            let digit = n % 10;
            if digit >= first {
                second = first;
                first = digit;
            } else if digit > second {
                second = digit;
            }
            n /= 10;
        }
        first * second
    }
}